integrations/langfuse/src/haystack_integrations/tracing/langfuse/tracer.py

```diff
@@ -287,6 +288,40 @@ def handle(self, span: LangfuseSpan, component_type: Optional[str]) -> None:
             coerced_output = tracing_utils.coerce_tag_value(span.get_data().get(_PIPELINE_OUTPUT_KEY))
             span.raw_span().update(input=coerced_input, output=coerced_output)
 
+        # special case for ToolInvoker (to update the span name to be: `original_component_name - [tool_names]`)
+        if component_type == "ToolInvoker":
+            tool_input_data = span.get_data().get("haystack.component.input")
+
+            tool_names = []
+            if tool_input_data and isinstance(tool_input_data.get("messages"), list):
+                for message in tool_input_data["messages"]:
+                    # should be true for all tool calls
+                    if isinstance(message, ChatMessage) and message.tool_calls:
+                        for tool_call in message.tool_calls:
+                            tool_names.append(tool_call.tool_name)
+
+            if tool_names:
+                try:
+                    original_name = span.get_data().get(_COMPONENT_NAME_KEY)
+                    if original_name:
```

Comment:
Just wondering, but if original_name doesn't exist wouldn't be better to fallback to using the `component_type` as the base name? So at the very least it we'd end up with `ToolInvoker - [...]` instead of no name?